### ISSUE TITLE
Training from presaved weights?

### ISSUE THREAD

**Issue by rsandler00** [larq/larq#465]
Hey, awesome package!

I know this isn't exactly an issue, but I wasn't sure where else to post (feel free to suggest somewhere).

I want to initialize by BNN w/ pretrained weights from an equivalent non binarized model. Currently I am using Larq layers in the BNN and standard tf.keras layers in the non-BNN. Restoring a checkpoint from the non-BNN obviously does not work. Is there a procedure you recommend for training a non-BNN and then initializing the BNN with (a binarized version of) the non-BNN weights?

(This might be good to place in the documentation somewhere)

Thanks, 
Roman

**Comment by lgeiger** [comment]
`larq` layers with `kernel_quantizer=None` and `input_quantizer=None` are equivalent to `tf.keras` layers. So if you use `larq` layers for the non-BNN as well, reloading with `model.load_weights` should work fine.

@timdebruin has more experience with pretraining, so he might have some more insights here.

**Comment by rsandler00** [comment]
Thanks for the quick response! 

Tangentially, I was wondering whether there is a group where I can post questions/issues I am having with binarizing my NN, which isnt a specific issue w/ the larq code (e.g. more theoretical questions). The larq git page seems to be focused on issues, and this may be too niche for stackoverflow

**Comment by lgeiger** [comment]
We have a community chat that you can join at https://spectrum.chat/larq, for questions and general discussions about BNNs.

**Comment by rsandler00** [comment]
> > Tangentially, I was wondering whether there is a group where I can post questions/issues I am having with binarizing my NN, which isnt a specific issue w/ the larq code (e.g. more theoretical questions and specific architecture/training procedure questions)
> 
> We have a community chat that you can join at https://spectrum.chat/larq, for questions and general discussions about BNNs.

Exactly what I saw looking for. Thanks!

**Comment by timdebruin** [comment]
Hi Roman,
I think @lgeiger answered the technical aspect already, I'll just add a small training tip:

Rather than using the weights of a  full-precision model to initialize the BNN, it can work quite nicely to use the weights of a binary activation network (BAN). To train a BAN simply give the `input_quantizer` argument you use for the BNN, but keep `kernel_quantizer=None`. Combine that with some weight decay or l2 regularization on those kernels for best effects. When subsequently training the BNN make sure to use a lower learning rate. 

I hope that helps, feel free to discuss more training details on spectrum and post technical questions here.
